### PR TITLE
[BANKCON-11768] Add E2E UI tests for Native Instant Debits flow

### PR DESCRIPTION
**Pull request by mats-stripe**
## Summary

This adds three simple end to end UI tests for the native instant debits flow. These test the "golden path" of these scenarios:

- New user experience
- Return user experience
- Account holder sign in experience

## Motivation

E2E test coverage!

## Testing

See code comments for videos of each test being executed!

## Changelog

N/a

### CODE REVIEW

File: Example/FinancialConnections Example/FinancialConnectionsUITests/InstantDebitsUITests.swift [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+import XCTest
+
+final class InstantDebitsUITests: XCTestCase {
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        continueAfterFailure = false
+    }
+
+    func test_nux() {
```
Comment (mats-stripe) [inline Example/FinancialConnections Example/FinancialConnectionsUITests/InstantDebitsUITests.swift:16]: https://github.com/user-attachments/assets/5acaab7c-12b8-436d-9aa2-b7f5d03bcc54

File: Example/FinancialConnections Example/FinancialConnectionsUITests/InstantDebitsUITests.swift [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        let featuredLegacyTestInstitution = app.tables.cells.staticTexts["Test OAuth Institution"]
+        XCTAssertTrue(featuredLegacyTestInstitution.waitForExistence(timeout: 60.0))
+        featuredLegacyTestInstitution.tap()
+
+        app.fc_nativePrepaneContinueButton.tap()
+        app.fc_nativeConnectAccountsButton.tap()
+        app.fc_nativeSuccessDoneButton.tap()
+    }
+
+    func test_rux() {
```
Comment (mats-stripe) [inline Example/FinancialConnections Example/FinancialConnectionsUITests/InstantDebitsUITests.swift:49]: https://github.com/user-attachments/assets/6659c02e-0726-4ea2-81af-136c113247dc

File: Example/FinancialConnections Example/FinancialConnectionsUITests/InstantDebitsUITests.swift [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+        let successBankAccountRow = app.staticTexts["Success"]
+        XCTAssertTrue(successBankAccountRow.waitForExistence(timeout: 60.0))
+        successBankAccountRow.tap()
+
+        app.fc_nativeConnectAccountsButton.tap()
+        app.fc_nativeSuccessDoneButton.tap()
+    }
+
+    func test_accountHolder() {
```
Comment (mats-stripe) [inline Example/FinancialConnections Example/FinancialConnectionsUITests/InstantDebitsUITests.swift:77]: https://github.com/user-attachments/assets/1c5602f7-3730-4218-b6f6-673655a4c297

File: StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/FinancialConnectionsAPIClient.swift [hunk 4]
```diff
@@ -379,7 +379,7 @@ extension FinancialConnectionsAPIClient: FinancialConnectionsAPI {
         return self.post(
             resource: APIEndpointAuthSessionsAuthorized,
             parameters: body,
-            useConsumerPublishableKeyIfNeeded: false
+            useConsumerPublishableKeyIfNeeded: true
```
Comment (mats-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/FinancialConnectionsAPIClient.swift:382]: Drive-by fix to these API requests - we need to use the consumer publishable keys for these!

**Comment by mats-stripe** [comment]
`PaymentSheetUITests.testInstantDebits_verticalmode`:

https://github.com/user-attachments/assets/e6a1f269-8bff-4f69-b06c-8dd62ec3abb0

File: StripeFinancialConnections/StripeFinancialConnections/Source/Common/FlowRouter.swift [hunk 5]
```diff
@@ -89,11 +89,6 @@ class FlowRouter {
     }
 
     private var shouldUseNativeInstantDebits: Bool {
-        // Show web instant debits flow while UITesting for now.
-        guard ProcessInfo.processInfo.environment["UITesting"] == nil else {
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Common/FlowRouter.swift:93]: it seems like you addressed this (I see changed in `PaymentSheetUITest`), but noting regardless that one thing to be careful is that we do have two tests written for instant debits in the payment code so have to be always be careful that they are not accidentally affected

**Review by kgaidis-stripe — DISMISSED** [review]
💯

File: StripeFinancialConnections/StripeFinancialConnections/Source/Common/FlowRouter.swift (hunk 5 above)
Comment (mats-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Common/FlowRouter.swift:93]: Yep! The `PaymentSheetUITests` now use the native instant debits flow, and there are still UI tests for the web instant debits flow in `FinancialConnectionsUITests`

**Review by kgaidis-stripe — APPROVED** [review]
